README.md

```diff
@@ -5,101 +5,159 @@
-
-# Syscall count by thread name
-bpftrace -e 'tracepoint:raw_syscalls:sys_enter { @[comm] = count(); }'
-
-# Read bytes by thread name:
-bpftrace -e 'tracepoint:syscalls:sys_exit_read /args.ret/ { @[comm] = sum(args.ret); }'
-
-# Read size distribution by thread name:
-bpftrace -e 'tracepoint:syscalls:sys_exit_read { @[comm] = hist(args.ret); }'
-
-# Show per-second syscall rates:
-bpftrace -e 'tracepoint:raw_syscalls:sys_enter { @ = count(); } interval:s:1 { print(@); clear(@); }'
-
-# Trace disk size by PID and thread name
-bpftrace -e 'tracepoint:block:block_rq_issue { printf("%d %s %d\n", pid, comm, args.bytes); }'
-
-# Count page faults by thread name
-bpftrace -e 'software:faults:1 { @[comm] = count(); }'
-
-# Count LLC cache misses by thread name and PID (uses PMCs):
-bpftrace -e 'hardware:cache-misses:1000000 { @[comm, pid] = count(); }'
 
-# Profile user-level stacks at 99 Hertz for PID 189:
-bpftrace -e 'profile:hz:99 /pid == 189/ { @[ustack] = count(); }'
+bpftrace provides a [Nix](https://nixos.org/nix/) [flake](https://wiki.nixos.org/wiki/Flakes), which is recommended for building and testing.
 
-# Files opened in the root cgroup-v2
-bpftrace -e 'tracepoint:syscalls:sys_enter_openat /cgroup == cgroupid("/sys/fs/cgroup/unified/mycg")/ { printf("%s\n", str(args.filename)); }'
+```bash
+nix develop
 ```
 
-More powerful scripts can easily be constructed. See [Tools](tools/README.md) for examples.
+For a suitable build environment without Nix, see our Dockerfiles for detailed build examples:
+- [Ubuntu](https://github.com/bpftrace/bpftrace/blob/master/docker/Dockerfile.ubuntu)
+- [Fedora](https://github.com/bpftrace/bpftrace/blob/master/docker/Dockerfile.fedora)
+- [Debian](https://github.com/bpftrace/bpftrace/blob/master/docker/Dockerfile.debian)
 
+If all dependencies are installed correctly, you should be able to configure and build using [CMake](https://cmake.org).
 
-## Support
-
-For additional help / discussion, please use our [discussions](https://github.com/bpftrace/bpftrace/discussions) page.
-
-We are also holding regular [office hours](https://docs.google.com/document/d/1nt010RfL4s4gydhCPSJ-Z5mnFMFuD4NrcpVmUcyvu2E/edit?usp=sharing) open to the public.
-
-## Probe types
-<center><a href="images/bpftrace_probes_2018.png"><img src="images/bpftrace_probes_2018.png" border=0 width=700></a></center>
-
-See the [Manual](man/adoc/bpftrace.adoc) for more details.
-
-## Plugins
-
-bpftrace has several plugins/definitions, integrating the syntax into your editor.
-<!--- Feel free to add your own plugins below, in alphabetical order -->
-
-- [Emacs](https://gitlab.com/jgkamat/bpftrace-mode)
-- [Vim](https://github.com/mmarchini/bpftrace.vim)
-- [VS Code](https://github.com/bolinfest/bpftrace-vscode)
-- [Bash Completion](https://github.com/scop/bash-completion)
+```bash
+cmake -DCMAKE_BUILD_TYPE=Release -B build .
+make -C build -j$(nproc)
+```
 
-## License
```

Comment:
If you're removing this from the README.md doesn't it need to go somewhere else and be linked to?